File: tools/unlzexe/sandbox-ghidra-gemini/dump_entry.py
# Dump entry point using Jython
# @category Sandbox

from ghidra.program.model.address import Address

def run():
    entry_sym = currentProgram.getSymbolTable().getPrimarySymbol("entry")
    if not entry_sym:
        print("No 'entry' symbol found!")
        return
    entry_addr = entry_sym.getAddress()
    print("Entry Point Address: {}".format(entry_addr))
    
    listing = currentProgram.getListing()
    ins_iter = listing.getInstructions(entry_addr, True)
    
    print("Disassembly starting at {}:".format(entry_addr))
    count = 0
    while ins_iter.hasNext() and count < 50:
        ins = ins_iter.next()
        print("  {}: {}".format(ins.getAddress(), ins))
        count += 1

if __name__ == '__main__':
    run()

File: tools/unlzexe/sandbox-ghidra-gemini/DumpEntry.java
//Dump entry point.
//@category Sandbox
//@keybinding 
//@menupath 
//@toolbar 

import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;
import ghidra.program.model.listing.Instruction;
import ghidra.program.model.listing.InstructionIterator;

public class DumpEntry extends GhidraScript {
    @Override
    public void run() throws Exception {
        Address entry = currentProgram.getSymbolTable().getPrimarySymbol("entry").getAddress();
        println("Entry Point Address: " + entry);
        dumpRange(entry, 0x100);
    }
    
    private void dumpRange(Address startAddr, int len) throws Exception {
        Address endAddr = startAddr.add(len);
        println("Disassembly from " + startAddr + " to " + endAddr + ":");
        InstructionIterator iter = currentProgram.getListing().getInstructions(startAddr, true);
        while (iter.hasNext()) {
            Instruction ins = iter.next();
            if (ins.getAddress().compareTo(endAddr) > 0) break;
            println("  " + ins.getAddress() + ": " + ins.toString());
        }
    }
}
